rate-limit/src/lib.rs

```diff
@@ -0,0 +1,436 @@
+
+			LimitWhitelist::<T>::try_mutate(rate_limiter_id, |whitelist| -> DispatchResult {
+				ensure!(whitelist.contains(&key_filter), Error::<T>::FilterNotExisted);
+				whitelist.remove(&key_filter);
+
+				Self::deposit_event(Event::WhitelistFilterRemoved { rate_limiter_id });
+				Ok(())
+			})
+		}
+
+		/// Resett whitelist filter rule.
+		///
+		/// Requires `GovernanceOrigin`
+		///
+		/// Parameters:
+		/// - `rate_limiter_id`: rate limiter id.
+		/// - `new_list`: the filter rule list to reset.
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::WeightInfo::reset_whitelist())]
+		#[transactional]
+		pub fn reset_whitelist(
+			origin: OriginFor<T>,
+			rate_limiter_id: T::RateLimiterId,
+			new_list: Vec<KeyFilter>,
+		) -> DispatchResult {
+			T::GovernanceOrigin::ensure_origin(origin)?;
+
+			let whitelist: BoundedVec<KeyFilter, T::MaxWhitelistFilterCount> =
+				BoundedVec::try_from(new_list).map_err(|_| Error::<T>::MaxFilterExceeded)?;
+			let ordered_set: OrderedSet<KeyFilter, T::MaxWhitelistFilterCount> = whitelist.into();
+			LimitWhitelist::<T>::insert(rate_limiter_id, ordered_set);
+
+			Self::deposit_event(Event::WhitelistFilterReset { rate_limiter_id });
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Access the RateLimitQuota, if RateLimitRule will produce new quota,
+		/// update RateLimitQuota and then return remainer_quota
+		pub fn access_remainer_quota_after_update(
+			rate_limit_rule: RateLimitRule,
+			limiter_id: &T::RateLimiterId,
+			encoded_key: &Vec<u8>,
+		) -> u128 {
+			RateLimitQuota::<T>::mutate(limiter_id, encoded_key, |(last_updated, remainer_quota)| -> u128 {
+				match rate_limit_rule {
+					RateLimitRule::PerBlocks { blocks_count, quota } => {
+						let now: u64 = frame_system::Pallet::<T>::block_number().saturated_into();
+						let interval: u64 = now.saturating_sub(*last_updated);
+						if interval >= blocks_count {
+							*last_updated = now;
+							*remainer_quota = quota;
+						}
+					}
+
+					RateLimitRule::PerSeconds { secs_count, quota } => {
+						let now: u64 = T::UnixTime::now().as_secs();
+						let interval: u64 = now.saturating_sub(*last_updated);
+						if interval >= secs_count {
+							*last_updated = now;
+							*remainer_quota = quota;
+						}
+					}
+
+					RateLimitRule::TokenBucket {
+						blocks_count,
+						quota_increment,
+						max_quota,
+					} => {
+						let now: u64 = frame_system::Pallet::<T>::block_number().saturated_into();
+						let interval: u64 = now.saturating_sub(*last_updated);
+						if !blocks_count.is_zero() && interval >= blocks_count {
+							let inc_times: u128 = interval
+								.checked_div(blocks_count)
+								.expect("already ensure blocks_count is not zero; qed")
+								.saturated_into();
+
+							*last_updated = now;
+							*remainer_quota = quota_increment
+								.saturating_mul(inc_times)
+								.saturating_add(*remainer_quota)
+								.min(max_quota);
+						}
+					}
+
+					_ => {}
```

Comment:
```suggestion
					Unlimited | NotAllowed  => {}
```

Will be good to explicitly list all cases.